### PR TITLE
refactor: passing network and deciding what url to use on wallet-service

### PR DESCRIPTION
**Pull request by andreabadesso**
# Acceptance criteria

The app must connect to the testnet wallet service instance when the network is `testnet`

### CODE REVIEW

File: src/wallet/api/walletServiceAxios.ts [hunk 1]
```diff
@@ -46,4 +51,17 @@ export const axiosInstance = async (wallet: HathorWalletServiceWallet | null = n
   return axios.create(defaultOptions);
 }
 
-export default axiosInstance;
\ No newline at end of file
+/**
+ * Returns the correct base url constant for wallet service based on the network
+ *
+ * @param {Network} network The network, can be either mainnet or testnet but will default to testnet
```
Comment (pedroferreira1) [inline src/wallet/api/walletServiceAxios.ts:57]: There is no default, the parameter is required, right?

File: src/wallet/api/walletServiceAxios.ts [hunk 2]
```diff
@@ -20,11 +25,11 @@ import HathorWalletServiceWallet from '../wallet';
  *
  * @param {number} timeout Timeout in milliseconds for the request
  */
-export const axiosInstance = async (wallet: HathorWalletServiceWallet | null = null, timeout: number = TIMEOUT) => {
+export const axiosInstance = async (wallet: HathorWalletServiceWallet | null = null, network: Network, timeout: number = TIMEOUT) => {
```
Comment (pedroferreira1) [inline src/wallet/api/walletServiceAxios.ts:28]: Should we refactor the parameters and set `network` as the first and only required parameter and add `wallet` and `timeout` as an optional options parameter just like we've been doing with optional parameters in the lib?

**Review by pedroferreira1 — CHANGES_REQUESTED** [review]

File: src/wallet/api/walletServiceAxios.ts (hunk 1 above)
Comment (andreabadesso) [reply src/wallet/api/walletServiceAxios.ts:57]: Yes.

What do you think about it?

File: src/wallet/api/walletServiceAxios.ts (hunk 2 above)
Comment (andreabadesso) [reply src/wallet/api/walletServiceAxios.ts:28]: I agree, done in 2b50b0be879e9da40172f4463b9e63dcd783a899

File: src/wallet/api/walletServiceAxios.ts (hunk 1 above)
Comment (pedroferreira1) [reply src/wallet/api/walletServiceAxios.ts:56]: I like it.

File: src/wallet/api/walletApi.ts [hunk 3]
```diff
@@ -8,6 +8,7 @@
 import { axiosInstance } from './walletServiceAxios';
 import Output from '../../models/output';
 import Input from '../../models/input';
+import Network from '../../models/network';
```
Comment (pedroferreira1) [inline src/wallet/api/walletApi.ts:11]: Are we using this?

**Review by pedroferreira1 — APPROVED** [review]

File: src/constants.js [hunk 4]
```diff
@@ -226,7 +226,8 @@ export const SELECT_OUTPUTS_TIMEOUT = 1000 * 60;
 /**
  * Wallet service URL
  */
-export const WALLET_SERVICE_BASE_URL  = 'https://wallet-service.testnet.hathor.network/';
+export const WALLET_SERVICE_BASE_URL  = 'https://wallet-service.hathor.network/';
```
Comment (msbrogli) [inline src/constants.js:229]: Shouldn't we use `wallet-service.mainnet.hathor.network`? I'm not sure what strategy we've been using.

**Review by msbrogli — APPROVED** [review]

File: src/constants.js (hunk 4 above)
Comment (andreabadesso) [reply src/constants.js:229]: I think this is a valid discussion, @luislhl probably followed the explorer URLs pattern:

https://explorer.hathor.network/ for `mainnet`
https://explorer.testnet.hathor.network/ for `testnet`

I like the current URL, without `.mainnet` but we can discuss this in another issue. What do you think?

**Pull request merged**